parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -17,30 +17,30 @@ const MINIMUM_DEPOSIT: u128 = 1;
 /// we may want to evolve the protocol so that the ethereum side sends XCM messages directly.
 /// Instead having BridgeHub transcode the messages into XCM.
 #[derive(Clone, Encode, Decode, RuntimeDebug)]
-pub enum VersionedMessage {
-	V1(MessageV1),
+pub enum VersionedMessage<G> {
```

Comment:
I'm uncomfortable adding generic types to `VersionedMessage` which are only going to be used for conversion purposes. Its perhaps a sign that conversion is becoming complicated enough to justify some refactoring.

So instead, perhaps we should introduce a separate converter that implements `sp_runtime::traits::Convert<VersionedMessage, Xcm<()>>`. The converter can be parameterised with all the conversion config it requires.

Reply:
https://github.com/Snowfork/snowbridge/pull/961/commits/2bcd6bec5df6861c402a0bd19135ec329f27965c